### ISSUE TITLE
atom-typescript (1 deprecation)

### ISSUE THREAD

**Issue by norfeo87** [TypeStrong/atom-typescript#1317]
The contents of atom-text-editor elements are no longer encapsulated within a shadow DOM boundary. Please, stop using rootElement and access the editor contents directly instead.
Report Issue
HTMLElement.rootElement - C:\Users\nORMA\AppData\Local\atom\app-1.19.0\resources\app\src\text-editor-element.js:23:10
getFromShadowDom - C:\Users\nORMA\.atom\packages\atom-typescript\dist\main\atom\tooltipManager.js:18:19
Object.attach - C:\Users\nORMA\.atom\packages\atom-typescript\dist\main\atom\tooltipManager.js:38:18
TypescriptEditorPane.setupTooltipView - C:\Users\nORMA\.atom\packages\atom-typescript\dist\main\typescriptEditorPane.js:187:24
new TypescriptEditorPane - C:\Users\nORMA\.atom\packages\atom-typescript\dist\main\typescriptEditorPane.js:140:14
subscriptions.add.atom.workspace.observeTextEditors - C:\Users\nORMA\.atom\packages\atom-typescript\dist\main\atomts.js:90:24

**Comment by abhisanand** [comment]
having same issue after updated to atom 1.19.0

**Comment by ZuBB** [comment]
a bit oftop but: starting from latest release (1.19) autocompletion stopped to work for me. can it be somehow related?

**Comment by chrissm79** [comment]
looks like autocompletion stopped working for me as well after the update to 1.19, not sure if this is related to the deprecation or if it's a separate issue.

**Comment by ZuBB** [comment]
thanks @chrissm79 for sharing this. I thought I got this due to some local misconfiguration..

**Comment by ZuBB** [comment]
@guncha do you have any ideas why autocomplete stopped to work?

**Comment by scttcper** [comment]
related https://github.com/TypeStrong/atom-typescript/pull/1310

**Comment by alexeden** [comment]
Installed Atom 1.19 today and atom-typescript became effectively unusable. I failed to understand how this was possible, given that the 1.19 beta has been out for a while now. So this is what I've found after clicking through some issues and comments:

- [An issue for the underlying cause](https://github.com/TypeStrong/atom-typescript/issues/1294) of autocomplete not working was created nearly 2 months ago
- [A pull request](https://github.com/TypeStrong/atom-typescript/pull/1310) was opened 20 days ago by @lierdakil that fixes this issue, yet it remains open and unmerged.
- 2 weeks before that, [an issue was created](https://github.com/TypeStrong/atom-typescript/issues/1273) that warned of the upcoming deprecations in Atom that could potentially cause problems. 

I mention all of this because in @lierdakil's PR, he asked a question whose sentiment I've come to wonder about myself:
> @TypeStrong, so is this repo dead?

I use TS at work, and I'm also the last holdout dev that hasn't switched from Atom to VScode. I *really* don't wanna switch, but my editor allegiance is contingent on stable Typescript integration. So, @TypeStrong, what up with all this?

**Comment by lierdakil** [comment]
FWIW, there's https://github.com/atom/ide-typescript, but it's *very* alpha as of right now, and is not published (although it's easy enough to install packages directly from github in Atom, `apm install atom/ide-typescript`). Might be a viable alternative. Bear in mind it depends on [atom-ide-ui](https://atom.io/packages/atom-ide-ui)
Alternatively, anyone feeling adventurous can use a version with #1310 applied via `apm install lierdakil/atom-typescript`. Disclamer: I provide no support whatsoever.

**Comment by alexeden** [comment]
I can vouch for @lierdakil's version of the atom-typescript package. Everything seems to be working again in 1.19. 💯

**Comment by emvidi** [comment]
Yes with @lierdakil version is working. Should we keep this or uninstall once the issue is resolved?

**Comment by lierdakil** [comment]
`lierdakil/atom-typescript` is a band-aid, and should be treated as such. Once the official package (i.e. this repo) is updated, you should remove `lierdakil/atom-typescript` (listed under "git packages" in Atom interface) and reinstall the official package. Does that make sense?

**Comment by emvidi** [comment]
@lierdakil  yes now it does, thks!

**Comment by jcardenasp** [comment]
Hi @lierdakil thanks for your help, I installed your version and work properly for me, Thanks again.

**Comment by rainstormza** [comment]
@lierdakil it's work !! thank you so much 👍

**Comment by walleboom** [comment]
@lierdakil Thank you! 👍

**Comment by jonyeezs** [comment]
Has anyone tried https://github.com/atom/ide-typescript?

And how it differs from this atom-typescript setup?

I hate asking these kinda questions on someone's repo. But this in-activeness is a sad state. And like most here still want to use Atom.

I'm putting my hands up to help maintain (or atleast manage this insane issue log) this project ✋ @TypeStrong @guncha if they do want help.

**Comment by lierdakil** [comment]
@jonyeezs ide-typescript is... fine, for the most part. Autocompletion can be sluggish at times, but I think that was fixed (or at least band-aided) recently. What I found lacking when compared to atom-typescript is the ability to easily typecheck whole project (which in atom-typescript is one keystroke). It's possible that I didn't look hard enough, but that leads me to another point: feature discoverability is abysmal. ide-typescript basically just binds to atom-ide-ui, and finding a feature in atom-ide-ui is... let's say, harder than it should be. There are also a few nitpicks, e.g. usage highlights are on ctrl+mouseover instead of on text cursor, which is less than ideal for keyboard-oriented workflows, but those I can live with.
